bitnami/kafka/templates/_helpers.tpl

```diff
@@ -98,181 +79,181 @@ Return the proper Docker Image Registry Secret Names
+{{- $res = "true" -}}
+{{- end -}}
+{{- end -}}
+{{- if $res -}}
+{{- true -}}
 {{- end -}}
 {{- end -}}
 
 {{/*
-Return true if encryption via TLS for inter broker communication connections should be configured
+Return the Kafka SASL credentials secret
 */}}
-{{- define "kafka.interBroker.tlsEncryption" -}}
-{{- $tlsProtocols := list "tls" "mtls" "sasl_tls" -}}
-{{- if (has .Values.auth.interBrokerProtocol $tlsProtocols) -}}
-    {{- true -}}
+{{- define "kafka.saslSecretName" -}}
+{{- $secretName := .Values.sasl.existingSecret -}}
+{{- if $secretName -}}
+    {{- printf "%s" (tpl $secretName $) -}}
+{{- else -}}
+    {{- printf "%s-user-passwords" (include "common.names.fullname" .) -}}
 {{- end -}}
 {{- end -}}
 
 {{/*
-Return true if encryption via TLS should be configured
+Return true if a SASL credentials secret object should be created
 */}}
-{{- define "kafka.tlsEncryption" -}}
-{{- if or (include "kafka.client.tlsEncryption" .) (include "kafka.interBroker.tlsEncryption" .) (include "kafka.externalClient.tlsEncryption" .) -}}
+{{- define "kafka.createSaslSecret" -}}
+{{- $secretName := .Values.sasl.existingSecret -}}
+{{- if and (or (include "kafka.saslEnabled" .) (or .Values.zookeeper.auth.client.enabled .Values.sasl.zookeeper.user)) (empty $secretName) -}}
     {{- true -}}
 {{- end -}}
 {{- end -}}
 
 {{/*
-Return the type of listener
-Usage:
-{{ include "kafka.listenerType" ( dict "protocol" .Values.path.to.the.Value ) }}
+Return true if a TLS credentials secret object should be created
 */}}
-{{- define "kafka.listenerType" -}}
-{{- if eq .protocol "plaintext" -}}
-PLAINTEXT
-{{- else if or (eq .protocol "tls") (eq .protocol "mtls") -}}
-SSL
-{{- else if eq .protocol "sasl_tls" -}}
-SASL_SSL
-{{- else if eq .protocol "sasl" -}}
-SASL_PLAINTEXT
+{{- define "kafka.tlsSecretName" -}}
+{{- $secretName := .Values.tls.existingSecret -}}
+{{- if $secretName -}}
+    {{- printf "%s" (tpl $secretName $) -}}
+{{- else -}}
+    {{- printf "%s-tls" (include "common.names.fullname" .) -}}
 {{- end -}}
 {{- end -}}
 
 {{/*
-Return the protocol used with zookeeper
+Return true if a TLS credentials secret object should be created
 */}}
-{{- define "kafka.zookeeper.protocol" -}}
-{{- if and .Values.auth.zookeeper.tls.enabled .Values.zookeeper.auth.client.enabled .Values.auth.sasl.jaas.zookeeperUser -}}
-SASL_SSL
-{{- else if and .Values.auth.zookeeper.tls.enabled -}}
-SSL
-{{- else if and .Values.zookeeper.auth.client.enabled .Values.auth.sasl.jaas.zookeeperUser -}}
-SASL
-{{- else -}}
-PLAINTEXT
+{{- define "kafka.createTlsSecret" -}}
+{{- if and (include "kafka.sslEnabled" .) (empty .Values.tls.existingSecret) .Values.tls.autoGenerated -}}
+    {{- true -}}
 {{- end -}}
 {{- end -}}
 
 {{/*
-Return the Kafka JAAS credentials secret
+Return the Kafka TLS credentials secret
 */}}
-{{- define "kafka.jaasSecretName" -}}
-{{- $secretName := .Values.auth.sasl.jaas.existingSecret -}}
+{{- define "kafka.tlsPasswordsSecretName" -}}
```

Comment:
There are still references to the to the old `kafka.jaasSecretName` helper

Reply:
Nice catch, thanks!